### DISCUSSION TITLE
How to understand why ipsec doesn't make connection without SPI's

### DISCUSSION THREAD

**Discussion by killgapwalker** [strongswan/strongswan#387] [General]
Why SPI's is necesery? Why He doesnt encapsulate Traffic in Interface?

**Comment by Thermi** [comment]
https://tools.ietf.org/html/rfc2409
https://tools.ietf.org/html/rfc5996

**Reply by killgapwalker** [reply]
Can you answer for users?

**Reply by Thermi** [reply]
SPI (Security Parameter Index) are an integral part of IKE and IPsec. They identify a set of transformations with their algorithms, keying material, replay state, and other. These sets of transformations are IPsec security assocaitions. Before any traffic can be transmitted, the two peers first need to agree on the algorithms and the shared keys to use, and whether to use extended sequence numbers. The shared key has to be different between IPsec SAs so a single leaked key does not reveal all corresponding plaintext of a ciphertext transmitted using these IPsec SAs, and also has to be negotiated first.
Therefore IPsec SAs need to be negotiated first.
